src/main/kotlin/org/rust/ide/formatter/RustAlignmentStrategy.kt

```diff
@@ -0,0 +1,42 @@
+package org.rust.ide.formatter
+
+import com.intellij.formatting.Alignment
+import com.intellij.lang.ASTNode
+
+/**
+ * Mimics [com.intellij.formatting.alignment.AlignmentStrategy], but offers more flexible API
+ */
+interface RustAlignmentStrategy {
```

Comment:
I am also thinking about ditching `alignment.kt` and `indent.kt` and integrating their logic directly inside `AbstractRustFmtBlock`. Just take a look at [this monster](https://github.com/intellij-rust/intellij-rust/pull/413/files#diff-253bc0944bbecd6130053f584e283569R35)